server/controllers/finance/journal/index.js

```diff
@@ -251,6 +253,9 @@ function editTransaction(req, res, next) {
     .then((currentTransaction) => {
       const posted = currentTransaction[0].posted;
 
+      // bind the current transaction under edit as "oldTransaction"
+      oldTransaction = currentTransaction;
```

Comment:
It would be good to use one name here with a comment description of what this actually represents - throughout this code `oldTransaction` and `currentTransaction` are the same entity but it is hard to follow the logic. 

I think `currentTransaction` is misleading because it ambiguously refers to either the current journal transaction or the current edited transaction. `oldTransaction` may be more appropriate.